### PR TITLE
8342541: Exclude List/KeyEventsTest/KeyEventsTest.java from running on macOS

### PR DESCRIPTION
**Pull request by Renjithkannath**
Hi Reviewers,

I have added the test tag  **@requires os.family != "mac"** and updated section with new format. 
Removed OSX check condition form the code.

Please review and let me know your suggestions if any.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8342541](https://bugs.openjdk.org/browse/JDK-8342541): Exclude List/KeyEventsTest/KeyEventsTest.java from running on macOS (**Bug** - P4)

### Reviewers
 * [Alexey Ivanov](https://openjdk.org/census#aivanov) (@aivanov-jdk - **Reviewer**)
 * [Phil Race](https://openjdk.org/census#prr) (@prrace - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/21924/head:pull/21924` \
`$ git checkout pull/21924`

Update a local copy of the PR: \
`$ git checkout pull/21924` \
`$ git pull https://git.openjdk.org/jdk.git pull/21924/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 21924`

View PR using the GUI difftool: \
`$ git pr show -t 21924`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/21924.diff">https://git.openjdk.org/jdk/pull/21924.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/21924#issuecomment-2459512955)
</details>

### CODE REVIEW

File: test/jdk/java/awt/List/KeyEventsTest/KeyEventsTest.java [hunk 1]
```diff
@@ -50,6 +39,18 @@
 
 import jdk.test.lib.Platform;
 
+/*
+*  @test
+*  @key headful
```
Comment (aivanov-jdk) [inline test/jdk/java/awt/List/KeyEventsTest/KeyEventsTest.java:44]:
```suggestion
/*
 *  @test
 *  @key headful
```
Align asterisks to the asterisk on the opening line.

File: test/jdk/java/awt/List/KeyEventsTest/KeyEventsTest.java [hunk 2]
```diff
@@ -264,9 +265,6 @@ private void doTest() throws Exception {
         boolean isWin = false;
         if (Platform.isWindows()) {
             isWin = true;
```
Comment (aivanov-jdk) [inline test/jdk/java/awt/List/KeyEventsTest/KeyEventsTest.java:267]: Can be simplified to
```suggestion
        boolean isWin = Platform.isWindows();
```

**Review by aivanov-jdk — CHANGES_REQUESTED** [review]

**Review by aivanov-jdk — APPROVED** [review]

**Review by prrace — APPROVED** [review]

**Pull request closed without merging**